src/api/graphapi.h

```diff
@@ -0,0 +1,417 @@
+);
+
+RG_Info RG_Edge_getDestNode     // get edge destination node
+(
+    RG_Node *n,                 // node extracted
+    RG_Edge e                   // handle of edge to extract destination node from
+);
+
+RG_Info RG_Edge_getConnectingEdge   // get edge connecting src to dest
+(
+    RG_Node src,                    // edge source node
+    RG_Node dest,                   // edge destination node
+    const char *label,              // edge label
+    RG_Edge *e                      // handle to edge
+);
+
+//------------------------------------------------------------------------------
+// RG_Edge_setAttribute
+//------------------------------------------------------------------------------
+
+// Set a single scalar in an edge.
+
+RG_Info RG_Edge_setAttribute_BOOL
+(
+    RG_Edge e,              // handle of edge to modify
+    const char *name,       // name of attribute to set
+    const bool value        // scalar to assign
+) ;
+
+RG_Info RG_Edge_setAttribute_NUMERIC
+(
+    RG_Edge e,              // handle of edge to modify
+    const char *name,       // name of attribute to set
+    const double value      // scalar to assign
+) ;
+
+RG_Info RG_Edge_setAttribute_STRING
+(
+    RG_Edge e,              // handle of edge to modify
+    const char *name,       // name of attribute to set
+    const char *value       // scalar to assign
+) ;
+
+//------------------------------------------------------------------------------
+// RG_Edge_getAttribute
+//------------------------------------------------------------------------------
+
+// Extract a single attribute from an edge.
+// Returns RG_SUCCESS if attribute is present, and sets x to its value.
+// Returns RG_NO_VALUE if attribute is not present, and x is unmodified.
+
+RG_Info RG_Edge_getAttribute_BOOL
+(    
+    bool *value             // scalar extracted
+    RG_Edge e,              // handle of edge to extract attribute from
+    const char *name        // name of attribute to get
+) ;
+
+RG_Info RG_Edge_getAttribute_NUMERIC
+(    
+    double *value           // scalar extracted
+    RG_Edge e,              // handle of edge to extract attribute from
+    const char *name        // name of attribute to get
+) ;
+
+RG_Info RG_Edge_getAttribute_STRING
+(    
+    char **value            // scalar extracted
+    RG_Edge e,              // handle of edge to extract attribute from
+    const char *name        // name of attribute to get
+) ;
+
+//------------------------------------------------------------------------------
+// RG_Edge_delete
+//------------------------------------------------------------------------------
+
+RG_Info RG_Edge_delete
+(
+    RG_Edge *n              // handle of edge to delete
+) ;
+
+
+//==============================================================================
+//=== RedisGraph Traversal API =================================================
+//==============================================================================
+
```

Comment:
It would be awesome if we could make the traversal API rich enough to support regular path expressions (Cypher doesn't yet satisfy all the requirements of regular path queries). The relationship patterns that would require is described well here: https://github.com/opencypher/openCypher/issues/179